packages/web-components/src/components/TrendChart/TrendChart.tsx

```diff
@@ -0,0 +1,73 @@
+import { ResponsiveLine } from "@nivo/line";
+import type { Point } from "@nivo/line";
+import type { FunctionalComponent } from "preact";
+import { useCallback } from "preact/hooks";
+import { defaultTrendChartConfig } from "./config";
+import * as styles from "./styles.scss";
+import { TrendChartKind } from "./types";
+import type { MeshTrendChartProps, Slice, SlicesTrendChartProps, TrendChartProps } from "./types";
+import { getKindConfig } from "./utils";
+
+export const TrendChart: FunctionalComponent<TrendChartProps> = ({
+  kind = TrendChartKind.Mesh,
+  width = 600,
+  height = 400,
+  emptyLabel = "No data available",
+  emptyAriaLabel = "No data available",
+  rootAriaLabel,
+  ...restProps
+}) => {
+  const kindConfig = getKindConfig(kind);
+
+  const handleClick = useCallback(
+    (data: Point | Slice, event: MouseEvent): void => {
+      if (kind === TrendChartKind.Mesh) {
+        return (restProps as MeshTrendChartProps)?.onClick?.(data as Point, event);
+      } else if ([TrendChartKind.SlicesX, TrendChartKind.SlicesY].includes(kind)) {
+        return (restProps as SlicesTrendChartProps)?.onSliceClick?.(data as Slice, event);
+      }
+    },
+    [kind, restProps],
+  );
+
+  const handleTouchEnd = useCallback(
+    (data: Point | Slice, event: TouchEvent): void => {
+      if (kind === TrendChartKind.Mesh) {
+        return (restProps as MeshTrendChartProps)?.onTouchEnd?.(data as Point, event);
```

Comment:
We don't need to return anything there because we just call functions in independent branches, so the can't overlap